pkg/ebpf/tracer.go

```diff
@@ -187,6 +221,208 @@ func (m *FlowTracer) Unregister() error {
 	return errors.New(`errors: "` + strings.Join(errStrings, `", "`) + `"`)
 }
 
+// Aggregates and consolidates the record values obtained the Per-CPU Hash Map
+func aggregateValues(mapValues []recordValue) recordValue {
+	var aggRecord recordValue
+	// Iterates over the array of record values and aggregates them into a single record
+	for _, mapValue := range mapValues {
+		aggRecord.Packets += mapValue.Packets
+		aggRecord.Bytes += mapValue.Bytes
+		if mapValue.FlowStartTime != 0 {
```

Comment:
Not really sure how this works, but should it be something like...?:

```go
if mapValue.FlowStartTime != 0 && mapValue.FlowStartTime < aggRecord.FlowStartTime {
```

Reply:
FlowStartTime is set only once

> I think there is something wrong in the way the packets are aggregated or flushed.
> 
> Experiment: I send an individual ICMP packet to a known IP; then I expect that it appears in the UI; then I send another packet. I'd expect that for each submission I get reported one small ongoing packet and one small returned incoming packet.
> 
> This is what I get with the old eBPF agent:
> 
> ![image](https://user-images.githubusercontent.com/939550/177165819-95c03e26-bba6-43ad-9c3a-fa9a4cb88fa4.png)
> 
> With this version, it seems that the packet numbers and sizes are somewhat random:
> 
> ![image](https://user-images.githubusercontent.com/939550/177165922-c563e27e-c792-4108-8e82-98ef1a99d8cf.png)

Thats weird, I tested with flowlogs-dump-collector, similar test, The agent is attached on a single interface.

```
$ go run flowlogs-dump-collector.go --listen_port=9999
starting flowlogs-dump-collector on port 9999
ipv4: 15:14:23.251708 ens6f0np0 IP 10.10.10.2:0 > 10.10.10.1:0: protocol:icmp dir:1 bytes:98 packets:1 ends: 15:14:23.251708
ipv4: 15:14:23.251983 ens6f0np0 IP 10.10.10.1:0 > 10.10.10.2:0: protocol:icmp dir:0 bytes:98 packets:1 ends: 15:14:23.251983
ipv4: 15:14:21.484647 ens6f0np0 IP 0.0.0.0:68 > 255.255.255.255:67: protocol:udp dir:0 bytes:333 packets:1 ends: 15:14:21.484647
arp: 15:14:28.458205 ens6f0np0 IP 0.0.0.0:0 > 0.0.0.0:0: protocol: dir:0 bytes:60 packets:1 ends: 15:14:28.458205
arp: 15:14:28.458234 ens6f0np0 IP 0.0.0.0:0 > 0.0.0.0:0: protocol: dir:1 bytes:42 packets:1 ends: 15:14:28.458234
ipv4: 15:14:39.142207 ens6f0np0 IP 10.10.10.2:0 > 10.10.10.1:0: protocol:icmp dir:1 bytes:98 packets:1 ends: 15:14:39.142207
ipv4: 15:14:39.142396 ens6f0np0 IP 10.10.10.1:0 > 10.10.10.2:0: protocol:icmp dir:0 bytes:98 packets:1 ends: 15:14:39.142396
```
The eviction from map timeout is 5 seconds, So you can expect multiple ICMP packets to be bunched if they are within that interval. However, in between the 98 byte packets, the other packets are weird. Do you see them in flowlogs-dump-collector as well?